### ISSUE TITLE
alluxio-fuse mounting hdfs fails: Mount point is a prefix of

### ISSUE THREAD

**Issue by Cameronwood611** [Alluxio/alluxio#16272]
**Alluxio Version:**
2.8.1

**Describe the bug**
Alluxio-Fuse fails when trying to mount hdfs

**To Reproduce**
TL;DR I went through these docs (in this order):
- https://docs.alluxio.io/os/user/stable/en/contributor/Building-Alluxio-From-Source.html
- https://docs.alluxio.io/os/user/stable/en/ufs/HDFS.html
- https://docs.alluxio.io/os/user/stable/en/compute/Tensorflow.html

Exact steps are as follows:
- clone repo (set ALLUXIO_HOME)
- mvn install -PhdfsActiveSync -Pufs-hadoop-3 -Dhadoop.version=3.3.4 -DskipTests
- cp conf/alluxio-site.properties.template conf/alluxio-site.properties
- Inside alluxio-site.properties:
   * `alluxio.master.mount.table.root.ufs=hdfs://<namenode>:<port>`
   * `alluxio.master.mount.table.root.option.alluxio.underfs.version=3.3.4`
- ./bin/alluxio format
- ./bin/alluxio-start.sh local
- ./bin/alluxio fs mkdir /training-data
- sudo mkdir -p /mnt/fuse
- sudo chown $(whoami) /mnt/fuse
- chmod 755 /mnt/fuse
- ./integration/fuse/bin/alluxio-fuse mount /mnt/fuse /training-data (stat command shows it's there)
- `./bin/alluxio fs mount /training-data/new/ hdfs://<namenode>:<port>/`

And that fails and outputs `Mount point hdfs://<namenode>:<port>/ is a prefix of hdfs://<namenode>:<port>/`.

**Expected behavior**
I expect Alluxio-fuse to mount hdfs correctly. Instead, /mnt/fuse is empty

**Urgency**
Eh 

**Are you planning to fix it**
I'm not planning to fix it

**Additional context**
I'm on Ubuntu 20.04

**Comment by ssz1997** [comment]
@Cameronwood611 Seems like your hdfs is already your root ufs, so your `fs mount` command is trying to mount your hdfs inside your hdfs. I don't think that's a valid operation.

If you want to access hdfs data from posix api, you don't need to do the last `fs mount` command. Without doing it, simply check if data exists in `/mnt/fuse`.

**Comment by Cameronwood611** [comment]
> @Cameronwood611 Seems like your hdfs is already your root ufs, so your `fs mount` command is trying to mount your hdfs inside your hdfs. I don't think that's a valid operation.
> 
> If you want to access hdfs data from posix api, you don't need to do the last `fs mount` command. Without doing it, simply check if data exists in `/mnt/fuse`.

Nope it's empty, `alluxio fs mount` does verify it's the root ufs though:  `<hdfs> on  /  (hdfs, not read-only, not shared, properties={alluxio.underfs.version=3.3.4})`

**Comment by ssz1997** [comment]
@Cameronwood611 Another thing I noticed is that the `/training-data` directory is newly created, so supposedly nothing is inside. You mounted this newly created dir to `/mnt/fuse` so there shouldn't be data inside. Try create a dir or file inside `/training-data` and check the mount point.

**Comment by Cameronwood611** [comment]
@ssz1997 Yes, creating a `test` directory within /training-data does show up in /mnt/fuse. I'm still not able to mount hdfs as it outputs the same error

**Comment by ssz1997** [comment]
@Cameronwood611  I think you already mount your hdfs to the root path of Alluxio? Since you already set your hdfs as `alluxio.master.mount.table.root.ufs`.

**Comment by Cameronwood611** [comment]
@ssz1997 Yes I believe that's how I can see everything on hdfs within the alluxio web ui. However, I'd like to be able to point to a path on my file system and pass that data to Docker. My thought was that I would have to use fuse mount and go through that Tensorflow document I referenced initially. I thought I was following the documentation in the correct order but what would be your suggestion?

**Comment by ssz1997** [comment]
@Cameronwood611 If your training data is inside hdfs, you can just use the mount point as a volume. The fs mount command in the doc is for letting Alluxio access the data. Your setting root ufs as your hdfs already served the purpose, so there's no need to mount again.

**Comment by Cameronwood611** [comment]
@ssz1997 Ah ok I understand. Since I could only get see hdfs with the command `alluxio fs ls /` I decided just to remove that mount and do it with fuse. So now I have everything at `/mnt/fuse/hdfs`. Thank you!
